### DISCUSSION TITLE
Enable draggable annotations

### DISCUSSION THREAD

**Discussion by stockiNail** [chartjs/chartjs-plugin-annotation#728] [Ideas]
We could happen that a user would like to move the annotations in an empty space on the chart in order to see the underlying data.

To enable that, we could add a new `draggable` option to the annotation that if is set to `true`, the user can drag the annotation (or annotations).

`mousedown` and `mouseup` event types should be added by the user in order to activate it.
The reset and the fallback to original position can be easily performed by a `chart.update()` to invoke externally (by a button or another controller).

**Comment by kurkle** [comment]
Should be almost as easy with an external plugin, so not sure if that is something that should be implementer. Always good to give examples though, so maybe a sample implementation instead?

**Reply by stockiNail** [reply]
I'm not sure if it could be easy because the plugin is not exposing the elements. I'll do an example in codepen soon.

**Reply by stockiNail** [reply]
Here codepen: https://codepen.io/stockinail/pen/OJOBEGm

You can drag the labels.

EDIT: added button to reset the drag.

**Comment by stockiNail** [comment]
@kurkle today I had time to think more about this enhancement. Well, finally I think you are right! This can be done by another plugin.
If you think it could be helpful, I'd like to prepare a codepen to show the implementation. And then, if you agree, I could add a sample in the interactions section. Let me know what you think.

**Comment by stockiNail** [comment]
Codepen (drag with external plugin) https://codepen.io/stockinail/pen/jOZPWXy
